graphqlapi/src/main/java/io/stargate/graphql/schema/DdlSchemaBuilder.java

```diff
@@ -453,6 +457,62 @@ private GraphQLFieldDefinition buildCreateTable() {
         .build();
   }
 
+  private GraphQLFieldDefinition buildCreateIndex() {
+    return GraphQLFieldDefinition.newFieldDefinition()
+        .name("createIndex")
+        .argument(GraphQLArgument.newArgument().name("keyspaceName").type(Scalars.GraphQLString))
+        .argument(
+            GraphQLArgument.newArgument().name("tableName").type(nonNull(Scalars.GraphQLString)))
+        .argument(
+            GraphQLArgument.newArgument().name("columnName").type(nonNull(Scalars.GraphQLString)))
+        .argument(GraphQLArgument.newArgument().name("indexName").type(Scalars.GraphQLString))
+        .argument(
+            GraphQLArgument.newArgument()
+                .name("indexType")
+                .description(
+                    "Adds a custom index type that can be identified by name (e.g., StorageAttachedIndex), "
+                        + "or class name (e.g., org.apache.cassandra.index.sasi.SASIIndex) ")
+                .type(Scalars.GraphQLString))
+        .argument(GraphQLArgument.newArgument().name("ifNotExists").type(Scalars.GraphQLBoolean))
+        .argument(
+            GraphQLArgument.newArgument()
+                .name("indexKeys")
+                .description("Index keys of a map")
+                .type(Scalars.GraphQLBoolean))
```

Comment:
A GraphQL enum might be a bit more user-friendly:
```
enum IndexKind {
  KEYS
  VALUES
  ENTRIES
  FULL
}
```
For the default kind we can either keep the field nullable, or introduce a `SIMPLE` constant (that's how it's called in the Cassandra codebase).

Reply:
Please, let me know if I have addressed this review comment accordingly. I am still getting acquainted with GraphQL. :)